### ISSUE TITLE
Game Details Name Search & Failed Metadata Download

### ISSUE THREAD

**Issue by kchau42** [JosefNemec/PlaynitePlugin-SteamStoreMetadata#10]
Hello there! Appreciate a lot of your work, I'm using your extension a lot for my offline Playnite library. :) Just wanted to report you some trouble that I have.

When I'm manually downloading the metadata, using the exact same name of the game, sometimes it doesn't find it. (I would say 50% of the time or more) Have to to put the game APP number (Don't have trouble with that).

Because it is not really reliable, have to search via google first.
Example : Game name "Salthe" https://store.steampowered.com/app/1357840/Salthe/ 

Second trouble : Sometimes the extension just doesn't do anything : https://store.steampowered.com/app/924520/Sunny_Shine_Funland/

Thank you for your time!

**Comment by edgar1016** [comment]
I noticed this as well with visual novels. Not sure if it's the issue but it seems to fail searching for games that have the 18+ content . Doesn't steam hide these games by default?

**Comment by JosefNemec** [comment]
If it's because they are game marked with age verification check, then there's nothing I can do about it. I can only add an ability to enter store URL into search box and find a game using that.

**Comment by shakeyourbunny** [comment]
> If it's because they are game marked with age verification check, then there's nothing I can do about it. I can only add an ability to enter store URL into search box and find a game using that.

Perhaps the scraper uses the login data of the steam library importer.

Another issue is that the scraper will fail not only after the adult gate because of the throttling by the Steam website; the scraper really should use the api key for a more generous limit.

Please also wait for 5 minutes if HTTP 249 comes, wait for the 5 minutes and then continue, but don't drop a failed condition.

**Comment by JosefNemec** [comment]
1) We already check for 429 error and add additional delay in that case https://github.com/JosefNemec/PlaynitePlugin-SteamStoreMetadata/blob/master/source/UniversalSteamMetadata/SteamShared/MetadataProvider.cs#L67
2) As far as I know, use of API key has no effect on rate limits, it only means that you can access resources that require certain privileges.
3) Data mining cookies (any data) from user's browser is a no go.
